contracts/vaults/RibbonThetaVault.sol

```diff
@@ -286,89 +268,66 @@ contract RibbonThetaVault is DSMath, OptionsVaultStorage {
      * @notice Deposits ETH into the contract and mint vault shares. Reverts if the underlying is not WETH.
      */
     function depositETH() external payable nonReentrant {
-        require(asset == WETH, "Not WETH");
-        require(msg.value > 0, "No value");
+        require(asset == WETH, "!WETH");
+        require(msg.value > 0, "!value");
 
-        IWETH(WETH).deposit{value: msg.value}();
         _deposit(msg.value);
+
+        IWETH(WETH).deposit{value: msg.value}();
     }
 
     /**
      * @notice Deposits the `asset` into the contract and mint vault shares.
      * @param amount is the amount of `asset` to deposit
      */
     function deposit(uint256 amount) external nonReentrant {
-        IERC20(asset).safeTransferFrom(msg.sender, address(this), amount);
         _deposit(amount);
+
+        IERC20(asset).safeTransferFrom(msg.sender, address(this), amount);
     }
 
     /**
      * @notice Mints the vault shares to the msg.sender
      * @param amount is the amount of `asset` deposited
      */
     function _deposit(uint256 amount) private {
-        uint256 totalWithDepositedAmount = totalBalance();
+        uint16 _round = round;
```

Comment:
where is `round` updated? should it be incremented in rollToNextOption?

Reply:
Yep, it will be incremented in rollToNextOption, it's in the next PR